tests/test_upload_counterpart_agreements.py

```diff
@@ -33,27 +45,174 @@ def test_upload_counterpart_file_uploads_and_calls_api_if_not_dry_run(getuser):
                 download_filename='The_supplier_who_signed-123456-agreement-countersignature.pdf'
             )
 
-            client.update_framework_agreement.assert_called_once_with(
+            data_api_client.update_framework_agreement.assert_called_once_with(
                 23,
-                {'countersignedAgreementPath':
+                {
+                    'countersignedAgreementPath':
                     'g-cloud-8/agreements/123456/123456-agreement-countersignature-2016-11-12-131415.pdf'
-                 },
+                },
                 'upload-counterpart-agreements script run by test user'
             )
 
 
-def test_upload_counterpart_file_does_not_upload_or_call_api_if_dry_run():
+def test_upload_counterpart_file_does_not_perform_actions_if_dry_run():
     bucket = mock.Mock()
-    client = mock.Mock()
-    client.get_supplier_framework_info.return_value = {
+    data_api_client = mock.Mock()
+    data_api_client.get_supplier_framework_info.return_value = {
         "frameworkInterest": {
             "agreementId": 23,
-            "declaration": {"nameOfOrganisation": "The supplier who signed"}
-        }
+            "declaration": {
+                "nameOfOrganisation": "The supplier who signed",
+                "primaryContactEmail": "supplier.primary@example.com",
+            },
+        },
+    }
+    data_api_client.find_users_iter.side_effect = lambda *args, **kwargs: iter((
+        {
+            "id": 111322,
+            "emailAddress": "user.111322@example.com",
+            "supplierId": 123456,
+            "active": True,
+        },
+        {
+            "id": 111321,
+            "emailAddress": "user.111321@example.com",
+            "supplierId": 123456,
+            "active": True,
+        },
+    ))
+    dm_notify_client = mock.Mock()
+    dm_notify_client.send_email.side_effect = AssertionError("This shouldn't be called")
```

Comment:
I have an intention to try and bring `assert_args_and_raise` and `assert_args_and_return` (as used in e.g. https://github.com/alphagov/digitalmarketplace-supplier-frontend/blob/9c5e42c6a227dd4bd287eacc231e88e93fdc2439/tests/app/main/test_frameworks.py#L49) in to dmutils for further use elsewhere.